File: dataset/human/java/file_9686.java
import java.util.Scanner;
public class Main {
private static Scanner s = new Scanner(System.in);

public static void main(String[] args) {
	 int n = s.nextInt();
     Node[] nodes = new Node[n];
     for (int i = 0; i < n; i++) {
         nodes[i] = new Node(i);
     }
     for (int i = 0; i < n; i++) {
         Node node = nodes[s.nextInt()];
         int left = s.nextInt();
         if (left != -1) node.setL(nodes[left]);
         int right = s.nextInt();
         if (right != -1) node.setR(nodes[right]);
         node.setSib();
     }
     Node root = null;
     for (int i = 0; i < n; i++) {
         if (nodes[i].parent == null) {
             root = nodes[i];
             break;
         }
     }
     root.setDep(0);
     for (int i = 0; i < n; i++) {
         nodes[i].setHei(0);
     }
     StringBuilder sb = new StringBuilder();
     for (int i = 0; i < n; i++) {
         sb.append("node ");
         sb.append(nodes[i].id);
         sb.append(": parent = ");
         if(nodes[i].parent == null) {
        	 sb.append(-1);
         }else {
        	 sb.append(nodes[i].parent.id);
         }
         sb.append(", sibling = ");
         if(nodes[i].sibling ==null) {
        	 sb.append(-1);
         }else {
        	 sb.append(nodes[i].sibling.id);
         }
         sb.append(", degree = ");
         sb.append(nodes[i].deg);
         sb.append(", depth = ");
         sb.append(nodes[i].dep);
         sb.append(", height = ");
         sb.append(nodes[i].hei);
         sb.append(", ");
         if (nodes[i].parent == null) {
             sb.append("root");
         } else {
             if (nodes[i].l == null && nodes[i].r == null) {
                 sb.append("leaf");
             } else {
                 sb.append("internal node");
             }
         }
         sb.append("\n");
     }
     System.out.print(sb);
 }
}


class Node{
	int id,deg,dep,hei;
	Node parent,sibling,l,r;

	Node(int id){
		this.id = id;
		this.deg = 0;
		this.dep = 0;
		this.hei = 0;
		}

	void setL(Node node) {
		l = node;
		l.parent = this;
		deg++;
	}

	void setR(Node node) {
		r = node;
		r.parent = this;
		deg++;
	}

	void setSib() {
		if(l != null && r != null) {
			l.sibling = r;
			r.sibling = l;
		}
	}

	void setDep(int dep){
		this.dep = dep;
		if(l != null)l.setDep(dep + 1);
		if(r != null)r.setDep(dep + 1);
	}

	void setHei(int hei) {
		if(hei >= this.hei) this.hei = hei;
		if(parent != null)parent.setHei(this.hei + 1);
	}

	}

File: dataset/human/c/file_9686.c
#include<stdio.h>
#include<stdlib.h>
#define MAX 100000

typedef struct{
  int parent;
  int child;
  int sibiling;
}Node;

int main(){
  Node T[MAX];
  int id,i,k,j=0,l=0,n,c;
  scanf("%d",&n);
  for(i=0;i<n;i++){
    T[i].parent=T[i].child=T[i].sibiling=-1;
  }
  for(i=0;i!=n;i++){
    scanf("%d%d",&id,&k);
    for(j=0;j<k;j++){
      scanf("%d",&c);
      if(T[id].child==-1){
      T[id].child=c;
      }else {
	T[l].sibiling=c;
      }
      T[c].parent=id;
      l=c;
    }
  }
  for(i=0;i<n;i++){
    j=i;
    c=0;
    while(T[j].parent!=-1){
      c++;
      j=T[j].parent;
    }
    printf("node %d: parent = %d, depth = %d, ",i,T[i].parent,c);
    if(T[i].parent==-1)printf("root, [");
    else if(T[i].child!=-1)printf("internal node, [");
    else if(T[i].child==-1)printf("leaf, [");
    if(T[i].child!=-1){
      j=T[i].child;
      printf("%d",T[i].child);
      while(T[j].sibiling!=-1){
	printf(", %d",T[j].sibiling);
	j=T[j].sibiling;
      }
     
    }
     printf("]\n");
  }
  return 0;
}
